File: src/candyshop/CandyShop.java
package CandyShop;

import java.util.ArrayList;

public class CandyShop {

    protected static String CANDY;
    protected static String LOLLIPOP;
    private int storedSugar;
    private double income;
    private ArrayList<Sweets> storage;

    public CandyShop(int sugar) {
        CANDY = "candy";
        LOLLIPOP = "lollipop";
        storage = new ArrayList<>();
        storedSugar = sugar;
    }

    public void createSweets(String name) {
        storage.add(new Sweets(name));
        storedSugar -= storage.get(storage.size() - 1).sugar;
    }

    public void raise(double percentage) {
        for (int i = 0; i < storage.size() ; i++) {
            storage.get(i).raisePrice(percentage);
        }
    }

    public void sell (String name, int sellCount) {
        while (sellCount > 0) {
            for (int i = 0; i < storage.size(); i++) {
                if (storage.get(i).name.equals(name)) {
                    income += storage.get(i).price;
                    storage.remove(i);
                    sellCount--;
                }
            }
        }
    }

    public void buySugar(int buyAmount) {
        storedSugar += buyAmount;
        income -= (buyAmount / 10);
    }

    public int countCandy() {
        int count = 0;
        for (int i = 0; i < storage.size(); i++) {
            if (storage.get(i).name.equals("candy")) {
                count++;
            }
        }
        return count;
    }

    public int countLollipop() {
        int count = 0;
        for (int i = 0; i < storage.size(); i++) {
            if (storage.get(i).name.equals("lollipop")) {
                count++;
            }
        }
        return count;
    }

    @Override
    public String toString() {
        return "Inventory: " + countCandy() + " candies, " + countLollipop() + " lollipops, Income: " +
                income + "$, Sugar: " + storedSugar + "gr";
    }
}

File: candyshop/candy_shop.py
# We run a candyshop.candyshop.Candy shop where we sell candies and lollipops
# One lollipop's price is 10$
# And it made from 5gr of sugar
# One candie's price is 20$
# And it made from 10gr of sugar
# we can raise their prices with a given percentage
#
# Create a candyshop.candyshop class
# It can store sugar and money as income. The constructor should take the amount of sugar in gramms.
# we can create lollipops and candies store them in the candyshop.candyshop's storage
# If we create a candie or lollipop the candyshop.candyshop's sugar amount gets reduced by the amount needed to create the sweets
# We can raise the prices of all candies and lollipops with a given percentage
# We can sell candie or lollipop with a given number as amount
# If we sell sweets the income will be increased by the price of the sweets and we delete it from the inventory
# We can buy sugar with a given number as amount. The price of 1000gr sugar is 100$
# If we buy sugar we can raise the candyshop.candyshop's amount of sugar and reduce the income by the price of it.
# The candyshop.candyshop should be represented as string in this format:
# "Inventory: 3 candies, 2 lollipops, Income: 100, Sugar: 400gr"

candy_shop = candyshop.candyshop(300)
candy_shop.create_sweets("candy")
candy_shop.create_sweets("candy")
candy_shop.create_sweets("lollipop")
candy_shop.create_sweets("lollipop")
print(candy_shop)
# Should print out:
# Invetory: 2 candies, 2 lollipops, Income: 0, Sugar: 270gr
candy_shop.sell("candy", 1)
print(candy_shop
# Should print out:
# "Invetory: 1 candies, 2 lollipops, Income:20, Sugar: 285gr"
candy_shop.raise_prices(5)
candy_shop.sell("lollipop", 1)
print(candy_shop)
# Should print out:
# "Invetory: 1 candies, 1 lollipops, Income:35, Sugar: 285gr"
candy_shop.buy_sugar(300)
print(candy_shop)
# Should print out:
# "Invetory: 1 candies, 1 lollipops, Income:5, Sugar: 315gr"
